### PR TITLE
usb: device: cdc_acm: add fifo size properties to CDC ACM UART bindings

### PR DESCRIPTION
**Pull request by jfischer-no**
Add RX and TX fifo size properties to CDC ACM UART bindings and
remove Kconfig option USB_CDC_ACM_RINGBUF_SIZE.
This change allow us finer configuration of CDC ACM UART per instance.
For most applications in the tree nothing changes as the
default value of the new properties is the same as default
value of the Kconfig option USB_CDC_ACM_RINGBUF_SIZE.

### CODE REVIEW

**Review by desowin — DISMISSED** [review]

**Review by anangl — DISMISSED** [review]

File: dts/bindings/serial/zephyr,cdc-acm-uart.yaml [hunk 1]
```diff
@@ -8,3 +8,18 @@ compatible: "zephyr,cdc-acm-uart"
 include: uart-controller.yaml
 
 on-bus: usb
+
+properties:
+    tx-fifo-size:
+      type: int
+      default: 1024
+      description: |
+        Size of the virtual CDC ACM UART TX FIFO
```
Comment (mbolivar-nordic) [inline dts/bindings/serial/zephyr,cdc-acm-uart.yaml:17]: You must justify why you chose both of these defaults in the description field here

File: doc/releases/release-notes-3.3.rst [hunk 2]
```diff
@@ -23,6 +23,10 @@ Changes in this release
 Removed APIs in this release
 ============================
 
+* Removed Kconfig option ``CONFIG_USB_CDC_ACM_RINGBUF_SIZE`` in favor of
+  devicetree properties tx-fifo-size and rx-fifo-size, which
+  allow us finer configuration of CDC ACM UART per instance.
```
Comment (mbolivar-nordic) [inline doc/releases/release-notes-3.3.rst:28]: Please add a reference to the compatible to the notes, so users know what you are talking about. This text is not very clear.

You can use this:

```
:dtcompatible:`zephyr,cdc-acm-uart`
```

to create a link to the compatible page in the bindings index  (https://docs.zephyrproject.org/latest/build/dts/api/bindings.html)

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]
2 minor requests from me. I think @galak might want to comment on whether we call these `tx-fifo-size` and `rx-fifo-size` or `zephyr,tx-fifo-size` / `zephyr,rx-fifo-size`.

File: dts/bindings/serial/zephyr,cdc-acm-uart.yaml (hunk 1 above)
Comment (jfischer-no) [reply dts/bindings/serial/zephyr,cdc-acm-uart.yaml:17]: I do not have good reason to justify it in the description, it does not make sense without knowing development history of CDC ACM UART implementation. But I mentioned it in the commit message, 
"...New properties takes the default
size value of USB_CDC_ACM_RINGBUF_SIZE which can be removed
subsequently."

**Pull request closed without merging**